src/main.rs

```diff
@@ -261,35 +264,131 @@ fn main() -> Result<(), StrErr> {
 	let mode = CompileMode::Check { test : false };
 	let compile_opts = args.compile_options(&config, mode, Some(&ws))?;
 
+	let metadata = {
+		let opts = OutputMetadataOptions {
+			features: vec![],
+			no_default_features: false,
+			all_features: false,
+			no_deps: false,
+			version: 1,
+		};
+		let metadata = cargo::ops::output_metadata(&ws, &opts)?;
+		serde_json::from_str::<Metadata>(&serde_json::to_string(&metadata)?)?
+	};
+
+	let metadata_packages = metadata
+		.packages
+		.iter()
+		.map(|p| (&p.id, p))
+		.collect::<HashMap<_, _>>();
+
+	let (dependency_names_by_lib_rename, dependency_names_by_lib_true_snakecased_name) = {
+		let current = ws.current()?.package_id();
+		let current = cargo_metadata::PackageId {
+			repr: format!(
+				"{} {} ({})",
+				current.name(),
+				current.version(),
+				current.source_id().into_url(),
+			),
+		};
+
+		let package = metadata_packages
+			.get(&current)
+			.unwrap_or_else(|| panic!("could not find {:?}", current.repr));
+
+		let renamed = package
+			.dependencies
+			.iter()
+			.flat_map(|d| d.rename.as_ref().map(|r| (r, d)))
+			.collect::<HashMap<_, _>>();
+
+		let unrenamed = package
+			.dependencies
+			.iter()
+			.filter(|d| d.rename.is_none())
+			.map(|d| (&d.name, d))
+			.collect::<HashMap<_, _>>();
+
+		let mut dependency_names_by_lib_rename = HashMap::new();
+		let mut dependency_names_by_lib_true_snakecased_name = HashMap::new();
+
+		for dep in metadata
```

Comment:
Can you change this to `metadata.resolve`? (resolve on same line). same for the other occurences.

Reply:
Changed.